Patch-Source: https://github.com/asottile/covdefaults/commit/18fbb656ef7e90ccd2af5dbef6f100054d96c866
---
From 18fbb656ef7e90ccd2af5dbef6f100054d96c866 Mon Sep 17 00:00:00 2001
From: Steve Kowalik <steven@wedontsleep.org>
Date: Mon, 31 Mar 2025 16:18:16 +1100
Subject: [PATCH] Support Coverage 7.7.0 and above

Coverage 7.7.0 refactored the way the Plugins class is used, and as a
consequence, the load_plugins method was renamed, and is no longer a
class method, instantiate the manager and load modules using the new
method name.
---
 tests/covdefaults_test.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/tests/covdefaults_test.py b/tests/covdefaults_test.py
index 2bf41c2..e71ef0f 100644
--- a/tests/covdefaults_test.py
+++ b/tests/covdefaults_test.py
@@ -228,7 +228,8 @@ def test_configure_keeps_existing_fail_under():
 
 def test_coverage_init():
     cfg = CoverageConfig()
-    plugin_manager = Plugins.load_plugins(['covdefaults'], cfg)
+    plugin_manager = Plugins()
+    plugin_manager.load_from_config(['covdefaults'], cfg)
     assert plugin_manager.get('covdefaults.CovDefaults')
 
 
